mage_ai/api/resources/PipelineRunResource.py

```diff
@@ -285,3 +285,10 @@ def update(self, payload, **kwargs):
             stop_pipeline_run(self.model, pipeline)
 
         return self
+
+    @safe_db_query
+    def delete(self, **kwargs):
```

Comment:
> Yes, it's fine. Did you check whether the associated block runs are deleted as well?

@wangxiaoyou1993  I had checked in the UI to make sure the block runs weren't appearing but didn't realize it was due to their `pipeline_run_ids` being set to `null`. Thanks for pointing this out. I added this `delete` method and tested it to make sure the pipeline run's associated block runs were being deleted (checked the actual db) when the pipeline run was deleted. Wasn't sure if unit tests were necessary for this.

Reply:
It'll be better to batch delete these models
```
        for br in block_runs:
            br.delete(commit=False)
        self.model.delete(commit=False)
        db_connection.session.commit()
```

Reply:
updated